urllib3/connectionpool.py

```diff
@@ -99,6 +99,30 @@ def close(self):
 _blocking_errnos = set([errno.EAGAIN, errno.EWOULDBLOCK])
 
 
+# The buffering argument to HTTPConnection.getresponse is only supported
+# as an implementation detail of CPython 2.7, so other versions and other
+# implementations don't support it.
+def _get_httplib_response(conn):
+    """Get HTTP connection response with buffering explicitly enabled"""
+    try:
+        # CPython 2.7 supports this call, while other versions and
+        # implementations may not do so (e.g. 3.x doesn't)
+        return conn.getresponse(buffering=True)
+    except TypeError:
+        pass
```

Comment:
This layout is to avoid Python 3's exception chaining if/when the conn.getresponse() call fails. However, I can bring the `_replace_get_httplib_response()` call inside the except block, and only have the `conn.getresponse()` call outside it, which will eliminate the need for `pass`.